assets/js/vueListComponent.js

```diff
@@ -0,0 +1,82 @@
+Vue.component('list', {
+    props: ["view", "filters"],
+    data() {
+        return {
+            data: [],
+            list: [],
+        }
+    },
+    methods: {
+        shuffle() {
+            // http://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm
+            for (var i = 0; i < this.data.length; i++) {
+                var j = i + Math.floor(Math.random() * (this.data.length - i));
+
+                var temp = this.data[j];
+                this.data[j] = this.data[i];
+                this.data[i] = temp;
+            }
+            this.list = this.data;
+        },
+        filterList() {
+            if (this.filters.type != "All") {
+                this.list = this.data.filter((itm) => {
+                    if (itm.type == null) {
```

Comment:
It is generally best to use `===` for equality checks, because the type of your values will also be checked.